### PR TITLE
fix: std/node/buffer - Buffer.copy doesn't work as expected

### PR DESCRIPTION
**Pull request by Soremwar**
Also added tests that represent the expected behaviour from Node

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @Soremwar, good fix and good tests

File: std/node/buffer.ts [hunk 1]
```diff
@@ -247,7 +253,10 @@ export default class Buffer extends Uint8Array {
     sourceStart = 0,
     sourceEnd = this.length,
   ): number {
-    const sourceBuffer = this.subarray(sourceStart, sourceEnd);
+    const sourceBuffer = this
+      .subarray(sourceStart, sourceEnd)
+      .subarray(0, Math.max(0, targetBuffer.length - targetStart));
+
     targetBuffer.set(sourceBuffer, targetStart);
```
Comment (devcat) [inline std/node/buffer.ts:260]: throws with `Buffer.alloc(5).copy(Buffer.alloc(0), 5)`

```suggestion
    if (sourceBuffer.length === 0) return 0;
    targetBuffer.set(sourceBuffer, targetStart);
```

**Review by devcat — CHANGES_REQUESTED** [review]

File: std/node/buffer.ts (hunk 1 above)
Comment (bartlomieju) [reply std/node/buffer.ts:260]: Thanks for the catch @devcat

**Review by bartlomieju — DISMISSED** [review]
Marking as `Request changes` based on feedback from @devcat

File: std/node/buffer.ts (hunk 1 above)
Comment (Soremwar) [reply std/node/buffer.ts:260]: Resolved and added test, thanks @devcat

**Comment by Soremwar** [comment]
@bartlomieju Corrected and ready for review

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**